File: Spring/ParkingStatus/src/main/java/com/ParkingStatus/ParkingStatus/Service/LotService.java
package com.ParkingStatus.ParkingStatus.Service;

import com.ParkingStatus.ParkingStatus.DataAccessService.Lot.LotDataAccessService;
import com.ParkingStatus.ParkingStatus.Models.Lot.Lot;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class LotService {

    @Autowired
    private final LotDataAccessService lotDataAccessService;

    @Autowired
    public LotService(LotDataAccessService lotDataAccessService) {

        this.lotDataAccessService = lotDataAccessService;
    }


    public List<Lot> getAllLots(){
//        lotDataAccessService.selectAllLots().forEach(System.out::println);
        return lotDataAccessService.selectAllLots();
    }

    public Lot getLotByID(int id){
        return lotDataAccessService.selectLotById(id);
    }

    public int createLot(Lot lot){

        return lotDataAccessService.insertLot(lot);
    }

    public int updateLot(int id, Lot lot){
        return lotDataAccessService.updateLot(id, lot);
    }

    public int deleteLot(int id){
        return lotDataAccessService.removeLot(id);
    }
}

File: Spring/ParkingStatus/src/js/parkingstatus.web/src/services/LotService.ts
import axios from 'axios';

const LOTS_REST_API_URL : string = 'http://localhost:8080/api/v1/lots';

class LotService{


    async getAllLots(){
        return axios.get(LOTS_REST_API_URL);
    }
    
    async getLotById(id :number){
        return axios.get(LOTS_REST_API_URL+'/'+ id);
    }

    async createLot(lot: any){
        return axios.post(LOTS_REST_API_URL, lot)
    }

    async updateLot(id: number, lot: any){
        return axios.post(LOTS_REST_API_URL+'/'+ id, lot);
    }

    deleteLot(id: number){
        return axios.delete(LOTS_REST_API_URL + '/'+id);
    }

}

export default new LotService();